lang/python/python-yaml/Makefile

```diff
@@ -8,46 +8,59 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=PyYAML
-PKG_VERSION:=3.13
+PKG_VERSION:=5.1
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://pypi.python.org/packages/source/P/PyYAML
-PKG_HASH:=3ef3092145e9b70e3ddd2c7ad59bdd0252a94dfe3949721633e41344de00a6bf
+PKG_HASH:=436bc774ecf7c103814098159fbb84c2715d25980175292c648f2da143909f95
 
 PKG_LICENSE:=MIT
-PKG_LICENSE_FILES:=COPYING
-PKG_MAINTAINER:=Nicolas Thill <nico@openwrt.org>
+PKG_LICENSE_FILES:=LICENSE
+PKG_CPE_ID:=cpe:/a:pyyaml_project:pyyaml
+PKG_MAINTAINER:=Josef Schlehofer <josef.schlehofer@nic.cz>
 
 PKG_BUILD_DIR:=$(BUILD_DIR)/$(BUILD_VARIANT)-$(PKG_NAME)-$(PKG_VERSION)
-PKG_UNPACK=$(HOST_TAR) -C $(PKG_BUILD_DIR) --strip-components=1 -xzf $(DL_DIR)/$(PKG_SOURCE)
 
 include $(INCLUDE_DIR)/package.mk
 include ../python-package.mk
 include ../python3-package.mk
 
+PKG_UNPACK=$(HOST_TAR) -C $(PKG_BUILD_DIR) --strip-components=1 -xzf $(DL_DIR)/$(PKG_SOURCE)
+
 define Package/python-yaml/Default
-  SUBMENU:=Python
   SECTION:=lang
+  SUBMENU:=Python
   CATEGORY:=Languages
-  URL:=https://pyyaml.org/wiki/PyYAML
+  TITLE:=YAML parser and emitter for Python
+  URL:=https://github.com/yaml/pyyaml
   DEPENDS:=+libyaml
 endef
 
 define Package/python-yaml
 $(call Package/python-yaml/Default)
-  TITLE:=YAML parser and emitter for Python
-  DEPENDS+=+PACKAGE_python-yaml:python-light +PACKAGE_python-yaml:python-codecs
+  DEPENDS+= \
+	+PACKAGE_python-yaml:python-light \
+	+PACKAGE_python-yaml:python-codecs
   VARIANT:=python
 endef
 
 define Package/python3-yaml
 $(call Package/python-yaml/Default)
-  TITLE:=YAML parser and emitter for Python3
   DEPENDS+=+PACKAGE_python3-yaml:python3-light
   VARIANT:=python3
 endef
 
+define Package/python-yaml/description
+PyYAML is a YAML parser and emitter for the Python programming language.
+endef
+
+define Package/python3-yaml/description
+$(call Package/python-yaml/description)
+.
+(Variant for Python3)
+endef
+
 define PyBuild/Compile
 	$(call Build/Compile/PyMod,,\
 		--with-libyaml install --prefix="$(PKG_INSTALL_DIR)/usr" \
```

Comment:
Would you mind seeing if setting:

```
PYTHON_PKG_SETUP_ARGS:=--with-libyaml
PYTHON3_PKG_SETUP_ARGS:=--with-libyaml
```

can replace the custom `PyBuild/Compile` / `Py3Build/Compile`? If it works, great; if not, can keep as is.

Reply:
When I used what you suggested to me, I'm getting the following error:

```
./setup.py  install --prefix="/usr" --root="/home/pepe/work/turrisos40/mox-hbd/build/build_dir/target-aarch64_cortex-a53_musl/python-PyYAML-5.1/ipkg-install" --with-libyaml ;
/home/pepe/work/turrisos40/mox-hbd/build/staging_dir/target-aarch64_cortex-a53_musl/usr/lib/python2.7/distutils/dist.py:267: UserWarning: Unknown distribution option: 'python_requires'
  warnings.warn(msg)
running install
error: error in command line: command 'install' has no such option 'with_libyaml'
make[2]: *** [Makefile:73: /home/pepe/work/turrisos40/mox-hbd/build/build_dir/target-aarch64_cortex-a53_musl/python-PyYAML-5.1/.built] Error 1

```

Reply:
> error: error in command line: command 'install' has no such option 'with_libyaml'

The hyphen in `with-libyaml` was converted into an underscore?

Reply:
Thanks for looking into this! I will edit it and see if it works. If yes, I will force push it and merge it.

Reply:
Since you added PKG_SETUP_GLOBAL_ARGS, it works when in Makefile, I have these 4 rows:

```
PYTHON_PKG_SETUP_GLOBAL_ARGS:=--with-libyaml
PYTHON_PKG_SETUP_ARGS:=
PYTHON3_PKG_SETUP_GLOBAL_ARGS:=--with-libyaml
PYTHON3_PKG_SETUP_ARGS:=
```

When, there is no PKG_SETUP_ARGS defined it ends up with following error:
`error: option --single-version-externally-managed not recognized`

Reply:
That's (somewhat) expected, as `PYTHON[3]_PKG_SETUP_ARGS` is set to `--single-version-externally-managed` by default.

Reply:
OK. Merging.